frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/common-templates-validations.ts

```diff
@@ -0,0 +1,82 @@
+import { getFieldId } from '../../../utils/vm-settings-tab-utils';
+import { iGetIn } from '../../../../../utils/immutable';
+import { UpdateOptions } from '../../types';
+import { iGetVmSettingAttribute } from '../../../selectors/immutable/vm-settings';
+import { iGetLoadedCommonData } from '../../../selectors/immutable/selectors';
+import { VMSettingsField, VMWizardProps, VMSettingsRenderableField } from '../../../types';
+import {
+  TEMPLATE_TYPE_LABEL,
+  TEMPLATE_WORKLOAD_LABEL,
+  TEMPLATE_OS_LABEL,
+  TEMPLATE_FLAVOR_LABEL,
+} from '../../../../../constants/vm';
+import { CommonTemplatesValidation } from './validation-types';
+
+const IDToJsonPath = {
+  [getFieldId(VMSettingsField.MEMORY)]: '.spec.domain.resources.requests.memory',
+  [getFieldId(VMSettingsField.CPU)]: '.spec.domain.cpu.cores',
+};
+
+export const getTemplateValidations = (
+  options: UpdateOptions,
+  fieldId: VMSettingsRenderableField,
+): CommonTemplatesValidation[] => {
+  const { getState, id } = options;
+  const state = getState();
+
+  // Get OS, workload-profile and flavor attributes
+  const os = iGetVmSettingAttribute(state, id, VMSettingsField.OPERATING_SYSTEM);
+  const flavor = iGetVmSettingAttribute(state, id, VMSettingsField.FLAVOR);
+  const workloadProfile = iGetVmSettingAttribute(state, id, VMSettingsField.WORKLOAD_PROFILE);
+
+  // Get all the templates from common-templates
+  const commonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+
+  let relevantTemplates;
+  // Get templates based on OS and WorkloadProfile
+  if (os && workloadProfile) {
+    relevantTemplates = (commonTemplates || []).filter(
```

Comment:
can you try to make  "getRelevantTemplates" method and use it here instead to doing it inline, i think it will look nicer ?